packages/kilt-did/src/types.chain.ts

```diff
@@ -0,0 +1,132 @@
+import type {
+  BTreeMap,
+  BTreeSet,
+  Enum,
+  Option,
+  Struct,
+  u64,
+  u8,
+  Vec,
+} from '@polkadot/types'
+import type {
+  AccountId,
+  BlockNumber,
+  Call,
+  Hash,
+} from '@polkadot/types/interfaces'
+
+export interface DidVerificationKey extends Enum {
+  isEd25519: boolean
+  isSr25519: boolean
+  asEd25519: Vec<u8>
+  asSr25519: Vec<u8>
+  type: 'sr25519' | 'ed25519'
+  value: Vec<u8>
+}
+
+export interface DidEncryptionKey extends Enum {
+  isX25519: boolean
+  asX25519: Vec<u8>
+  type: 'x25519'
+  value: Vec<u8>
+}
+
+export interface VerificationKeyDetails extends Struct {
+  /// A verification key the DID controls.
+  verificationKey: DidVerificationKey
+  /// The block number in which the verification key was added to the DID.
+  blockNumber: BlockNumber
+}
+
+export type KeyId = Hash
+
+export type DidIdentifier = AccountId
+
+export interface UrlEncoding extends Struct {
+  payload: Vec<u8>
+}
+
+export interface Url extends Enum {
+  isHttp: boolean
+  isFtp: boolean
+  isIpfs: boolean
+  asHttp: UrlEncoding
+  asFtp: UrlEncoding
+  asIpfs: UrlEncoding
```

Comment:
Don't we have `type` and `value` here as well?

Reply:
I don't see it either in other enums after, so perhaps it is just not required everywhere 😄

Reply:
that's coming from extending `Enum`, it inherits the properties defined there